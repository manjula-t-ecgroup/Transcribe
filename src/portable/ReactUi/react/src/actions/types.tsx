export const FETCH_USERS = 'FETCH_USERS';
export const SELECT_USER = 'SELECT_USER';