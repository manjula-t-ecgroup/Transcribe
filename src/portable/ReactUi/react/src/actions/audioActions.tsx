import Axios from 'axios';
import { COPY_AUDIO, INITIAL_TRANSCRIPTION, JUMP_CHANGE, PLAY_STATUS, PLAYSPEEDRATE_CHANGE,
    REPORT_POSITION, REQUEST_POSITION, SAVE_STATUS, SAVE_TOTAL_SECONDS, SUBMIT_STATUS } from './types';


export function playStatus(playing: boolean): any{
    return {
        payload: playing,
        type: PLAY_STATUS
    }
}

export function playSpeedRateChange(playSpeedRate: number): any{
    return {
        payload: playSpeedRate,
        type: PLAYSPEEDRATE_CHANGE
    }
}

export function jumpChange(jump: number): any{
    return {
        payload: jump,
        type: JUMP_CHANGE
    }
}

export function saveTotalSeconds(seconds: number): any {
    return {
        payload: seconds,
        type: SAVE_TOTAL_SECONDS
    }
}

export const reportPosition = (taskid: string, seconds: number) => (dispatch: any) => {
    dispatch({payload: seconds, type: REPORT_POSITION});
    Axios.put('/api/ReportPosition?task=' + taskid + '&position=' + seconds.toString());
}

export function requestPosition(): any {
    return {
        type: REQUEST_POSITION
    }
}

export function setInitialTranscription(status: boolean): any {
    return {
        payload: status,
        type: INITIAL_TRANSCRIPTION
    }
}

export function setSubmitted(submit: boolean): any{
    return {
        payload: submit,
        type: SUBMIT_STATUS
    }
}

export function setSaved(saved: boolean): any{
    return {
        payload: saved,
        type: SAVE_STATUS
    }
}

export const copyAudio = (task: string, project: string, audioFile: string, data: object) => (dispatch: any) => {
    dispatch({type: COPY_AUDIO});
    Axios.put('/api/CopyAudio?task=' + task + '&project=' + project+ '&audioFile=' + audioFile, data)
}