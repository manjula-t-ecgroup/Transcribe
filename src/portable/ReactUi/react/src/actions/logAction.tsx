import Axios from 'axios';

export const log = (message: string) => {
    Axios.put('/api/Log?' + message);
}
