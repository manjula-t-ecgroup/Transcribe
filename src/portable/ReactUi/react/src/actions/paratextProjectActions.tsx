import Axios from 'axios';
import { FETCH_PARATEXT_PROJECTS, SELECT_PARATEXT_PROJECT } from './types';

export const fetchParatextProjects = () => (dispatch: any) => {
    // tslint:disable-next-line:no-console
    console.log("in get")
    Axios.get('/api/GetParatextProjects').
        then(paratextProjects => {
            dispatch({
                payload: paratextProjects,
                type: FETCH_PARATEXT_PROJECTS
            });
        });
}

export function selectParatextProject(id: string): any{
    return {
        payload: id,
        type: SELECT_PARATEXT_PROJECT
    }
}