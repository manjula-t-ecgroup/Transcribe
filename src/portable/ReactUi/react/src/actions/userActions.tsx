import Axios from 'axios';
import { FETCH_USERS, SELECT_USER, UPDATE_AVATAR, UPDATE_USER_PENDING } from './types';


export const fetchUsers = () => (dispatch: any) => {
    Axios.get('/api/GetUsers').
        then(users => {
            dispatch({
                payload: users,
                type: FETCH_USERS
            });
        });
}

export function selectUser(id: string): any{
    return {
        payload: id,
        type: SELECT_USER
    }
}

export const updateUser = (user: string, project: string, query: string) => (dispatch: any) => {
    dispatch({type: UPDATE_USER_PENDING});
    Axios.put('/api/UpdateUser?user=' + user + '&project=' + project + query)
        .then(dispatch(fetchUsers()))
}

export const updateAvatar = (user: string, data: object) => (dispatch: any) => {
    dispatch({type: UPDATE_AVATAR});
    Axios.put('/api/UpdateAvatar?user=' + user, data)
        .then(dispatch(fetchUsers()))
}