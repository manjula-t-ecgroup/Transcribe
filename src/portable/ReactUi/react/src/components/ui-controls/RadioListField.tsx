import * as React from 'react';
import { connect } from 'react-redux';
import { IProjectSettingsStrings } from '../../model/localize';
import { IState } from '../../model/state';
import userStrings from '../../selectors/localize';
import PasswordField from './PasswordField';
import './RadioListField.sass';

interface IProps extends IStateProps {
    id?: string;
    options: string[];
    selected?: string;
    adminPassword?: string;
    message?: string;
    onChange?: (value: string) => any;
}

class RadioListField extends React.Component<IProps, any> {
    public state = {
        current: this.props.selected,
    }

    constructor(props: IProps) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }

    public handleChange(event: any) {
        this.setState({ ...this.state, current: event.target.value })
        if (this.props.onChange != null) {
            this.props.onChange(event.target.value)
        }
    }

    public handlePasswordChange(event: any) {
        if(event.target === undefined){ return;}
      this.setState({ ...this.state, password: event.target.value });
      if (this.props.onChange != null) {
        this.props.onChange(event.target.value);
      }
    }

    public render() {
        const { adminPassword, options, strings } = this.props
        const { current } = this.state
        const password  = (adminPassword !== undefined)? adminPassword: "";
        let currentValue = current;
        if (current !== undefined) {
            currentValue = current.toLowerCase();
        }
        const elementList = options.map((item: string) =>
            <div key={item}>
                <input
                    className="DataColor"
                    type="radio"
                    name="radioGroup"
                    value={item}
                    checked={currentValue === item.toLowerCase()}
                    onChange={this.handleChange} />
                <label className="labelRadioCaption">{item} </label>
                {(currentValue === "admin" && currentValue === item.toLowerCase()) ? (
                    <div className={"showPasswordField"}>
                        <PasswordField
                            caption={strings.passphrase}
                            id="password"
                            inputValue={password}
                        />
                    </div>
                ) : (
                        ""
                    )}
            </div>
        );

        return (
            <div className="RadioListField">
                <div className="bodyRow">
                    <div className="dataColumn">
                        <form>
                            <fieldset id="radioGroup" className="radioFieldSet">
                                {elementList}
                            </fieldset>
                        </form>
                    </div>
                </div>
            </div>
        )
    }
}

interface IStateProps {
    strings: IProjectSettingsStrings;
};

const mapStateToProps = (state: IState): IStateProps => ({
    strings: userStrings(state, { layout: "projectSettings" }),
});

export default connect(mapStateToProps)(RadioListField);