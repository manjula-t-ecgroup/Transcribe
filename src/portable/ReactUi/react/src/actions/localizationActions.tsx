import Axios from 'axios';
import { log } from '../actions/logAction';
import { FETCH_LOCALIZATION } from './types';


export const fetchLocalization = () => (dispatch: any) => {
    Axios.get('/localization/strings.json')
        .then(strings => {
            dispatch({
                payload: strings,
                type: FETCH_LOCALIZATION
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " fetch localization"))
        });
}
