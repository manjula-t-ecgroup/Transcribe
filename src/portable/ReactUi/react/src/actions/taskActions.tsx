import Axios from 'axios';
import { setSubmitted } from './audioActions';
import { ASSIGN_TASK_PENDING, COMPLETE_REVIEW_PENDING, COMPLETE_TRANSCRIPTION_PENDING, 
    FETCH_TASKS, FETCH_TRANSCRIPTION, SELECT_POPUP_TASK, SELECT_PROJECT, SELECT_TASK,  UNASSIGN_TASK_PENDING,
    UPDATE_TASK, WRITE_FULFILLED, WRITE_PENDING } from './types';
import { saveUserSetting } from './userActions';

export const assignTask = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: ASSIGN_TASK_PENDING});
    Axios.put('/api/TaskEvent?action=Assigned&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
        .then(dispatch(setSubmitted(false)))
}

export const unAssignTask = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: UNASSIGN_TASK_PENDING});
    Axios.put('/api/TaskEvent?action=Unassigned&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const completeTranscription = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: COMPLETE_TRANSCRIPTION_PENDING});
    Axios.put('/api/TaskEvent?action=TranscribeEnd&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const completeReview = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: COMPLETE_REVIEW_PENDING});
    Axios.put('/api/TaskEvent?action=ReviewEnd&task=' + taskid + '&user=' + userid)
        .then(dispatch(uploadTranscription(taskid, userid)))
}

export const uploadTranscription = (taskid: string, userid: string) => (dispatch: any) => {
    Axios.put('/api/TaskEvent?action=Upload&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const writeTranscription = (taskid: string, length: number, lang: string, dir: string, data: any) => (dispatch: any) => {
    dispatch({type: WRITE_PENDING});
    Axios.put('/api/WriteTranscription?task=' + taskid + "&length=" + length.toString() + "&lang=" + lang + "&dir=" + dir, data)
        .then(dispatch({type: WRITE_FULFILLED}))
}

export const fetchTasks = (username: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?user=' + username)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_TASKS
            });
        })
}

export const fetchTasksOfProject = (projectname: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?project=' + projectname)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_TASKS
            });
            dispatch({
                payload: projectname,
                type: SELECT_PROJECT
            })
        })
    }

export function selectProject(id: string): any{
    return {
        payload: id,
        type: SELECT_PROJECT
    }
}

export const  selectTask = (user: string, id: string) => (dispatch:any) => {
    dispatch({
        payload: id,
        type: SELECT_TASK
    })
    dispatch(fetchTranscription(id));
    dispatch(saveUserSetting(user, "lastTask", id))
    dispatch(setSubmitted(false))
}

export const  selectPopupTask = (id: string) => (dispatch:any) => {
    dispatch({
        payload: id,
        type: SELECT_POPUP_TASK
    })
}

export const fetchTranscription = (taskid: string) => (dispatch: any) => {
    const part = taskid.split('.');
    Axios.get('/api/audio/' + part[0] + '.transcription').
    then(transcription => {
        dispatch({
            payload: transcription,
            type: FETCH_TRANSCRIPTION
        });
    });
}

export const updateTask = (task: string, project: string, query: string, data: object) => (dispatch: any) => {
    dispatch({type: UPDATE_TASK});
    Axios.put('/api/UpdateTask?task=' + task + '&project=' + project + query, data)
        .then(dispatch(fetchTasksOfProject(project)))
}
