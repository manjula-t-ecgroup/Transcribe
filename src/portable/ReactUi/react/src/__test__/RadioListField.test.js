/* jshint esversion: 6 */
import * as React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import * as actions from '../actions/audioActions';
import { RadioListField } from '../components/ui-controls/RadioListField';

configure({ adapter: new Adapter() })

// Snapshot for RadioListField
describe('>>>Control: RadioListField --- Snapshot',()=>{
    const minProps = {
        options: [ "admin", "reviewer", "transcriber"],
        strings: {
            passphrase: "Enter Password"
        }
    };

    it('+++capturing Snapshot of RadioListField', () => {
        const renderedValue =  renderer.create(<RadioListField {...minProps}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: RadioListField', () => {
    let wrapper;
    const minProps = {
        options: [ "admin", "reviewer", "transcriber"],
        strings: {
            passphrase: "Enter Password"
        }
    };

    beforeEach(()=>{
        wrapper = shallow(<RadioListField {...minProps} />)
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });
});
