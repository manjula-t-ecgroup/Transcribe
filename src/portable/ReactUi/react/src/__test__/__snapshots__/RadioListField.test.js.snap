// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: RadioListField --- Snapshot +++capturing Snapshot of RadioListField 1`] = `
<div
  className="RadioListField"
>
  <div
    className="bodyRow"
  >
    <div
      className="dataColumn"
    >
      <form>
        <fieldset
          className="radioFieldSet"
          id="radioGroup"
        >
          <div>
            <input
              checked={false}
              className="DataColor"
              name="radioGroup"
              onChange={[Function]}
              type="radio"
              value="admin"
            />
            <label
              className="labelRadioCaption"
            >
              admin
               
            </label>
            
          </div>
          <div>
            <input
              checked={false}
              className="DataColor"
              name="radioGroup"
              onChange={[Function]}
              type="radio"
              value="reviewer"
            />
            <label
              className="labelRadioCaption"
            >
              reviewer
               
            </label>
            
          </div>
          <div>
            <input
              checked={false}
              className="DataColor"
              name="radioGroup"
              onChange={[Function]}
              type="radio"
              value="transcriber"
            />
            <label
              className="labelRadioCaption"
            >
              transcriber
               
            </label>
            
          </div>
        </fieldset>
      </form>
    </div>
  </div>
</div>
`;
